ouroboros-consensus/test-storage/Test/Ouroboros/Storage/FS/StateMachine.hs

```diff
@@ -1104,6 +1108,13 @@ tag = C.classify [
           Left TagPutSeekNegGet
         _otherwise -> Right $ tagPutSeekNegGet put seek
 
+    tagExclusiveFail :: EventPred
```

Comment:
I think this tagging isn't quite right. We don't want to read, and we want to check the error. I recommend looking at the minimal example for this tag: we're expecting to see a program that creates a file, closes it, and then tries to open it for writing with exclusive mode.

Reply:
@edsko right, that's a mistake. Changing to this:
```
tagExclusiveFail :: EventPred
    tagExclusiveFail = C.predicate $ \ev ->
      case (eventMockCmd ev, eventMockResp ev) of
        (Open _ _ True, Resp (Left fsError)) | fsErrorType fsError == FsResourceAlreadyExist->
          Left TagExclusiveFail
        _otherwisw -> Right tagExclusiveFail
```
gives the correct minimal counter example
```
do
  [x0,x1] <- open "/t1" a False
  close x1
  open x0 a True
```

Reply:
Perfect :)